lambdaNotes/src/components/DisplayNoteList.js

```diff
@@ -0,0 +1,62 @@
+import DisplayNoteCard from "./DisplayNoteCard";
+import React, { Component } from "react";
+import SearchNote from "./SearchNote";
+import { searchFunc } from "../util";
+
+// DisplayNoteList component is presentational component that manage display of list of note
+class DisplayNoteList extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      query: "",
+      isSearched: false
+    };
+
+    this.displayedNotes = [];
+  }
+
+  // to see if "View Your Notes" is clicked.  I use the props.isSearched boolean.
+  // Toggle isSearched in redux state to decide whether all notes should be displayed, or
+  // notes with search result should be displayed
+  // componentDidUpdate checks if isSearched has changed.
+  componentDidUpdate(prevProps) {
+    if (prevProps.isSearched !== this.props.isSearched) {
+      this.setState({
+        isSearched: this.props.isSearched,
+        query: ""
+      });
+    }
+  }
+
+  handleInputChange = e => {
+    this.setState({
+      [e.target.name]: e.target.value,
+      isSearched: true
+    });
+    this.props.handleSearchBoolean(true);
```

Comment:
[https://jaredpalmer.com/formik/docs/overview#why-not-redux-form](url)
[https://github.com/reduxjs/redux/issues/1287#issuecomment-175351978](url)

on every key stroke i am dispatching actions to my reducers. Seems kind of awkward & unnecessary to do this.
But either way in the end it works, so great job figuring out how to make it work. Functionally is great